rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/RDF4JTermFactory.java

```diff
@@ -0,0 +1,509 @@
+	 *            {@link Model} to ensure {@link BlankNode#equals(Object)} and
+	 *            {@link BlankNode#uniqueReference()} works as intended.
+	 * @param <T>
+	 *            The subclass of {@link Value}, e.g. {@link BNode}
+	 * @return A {@link RDFTerm} that corresponds to the RDF4J value
+	 * @throws IllegalArgumentException
+	 *             if the value is not a BNode, Literal or IRI
+	 */
+	@SuppressWarnings("unchecked")
+	public static <T extends Value> RDF4JTerm<T> asRDFTerm(final T value, UUID salt) {
+		if (value instanceof BNode) {
+			return (RDF4JTerm<T>) new BlankNodeImpl((BNode) value, salt);
+		}
+		if (value instanceof org.eclipse.rdf4j.model.Literal) {
+			return (RDF4JTerm<T>) new LiteralImpl((org.eclipse.rdf4j.model.Literal) value);
+		}
+		if (value instanceof org.eclipse.rdf4j.model.IRI) {
+			return (RDF4JTerm<T>) new IRIImpl((org.eclipse.rdf4j.model.IRI) value);
+		}
+		throw new IllegalArgumentException("Value is not a BNode, Literal or IRI: " + value.getClass());
+	}
+
+	/**
+	 * Adapt a RDF4J {@link Statement} as a Commons RDF {@link Triple}.
+	 * 
+	 * @param statement
+	 *            The statement to convert
+	 * @param salt
+	 *            A {@link UUID} salt to use for uniquely mapping any
+	 *            {@link BNode}s. The salt should typically be the same for
+	 *            multiple statements in the same {@link Repository} or
+	 *            {@link Model} to ensure {@link BlankNode#equals(Object)} and
+	 *            {@link BlankNode#uniqueReference()} works as intended.
+	 * @return A {@link Triple} that corresponds to the RDF4J statement
+	 */
+	public static RDF4JTriple asTriple(final Statement statement, UUID salt) {
+		return new TripleImpl(statement, salt);
+	}
+
+	private UUID salt = UUID.randomUUID();
+
+	private final ValueFactory valueFactory;
+
+	public RDF4JTermFactory() {
+		this.valueFactory = SimpleValueFactory.getInstance();
+	}
+
+	public RDF4JTermFactory(ValueFactory valueFactory) {
+		this.valueFactory = valueFactory;
+	}
+
+	/**
+	 * Adapt a RDF4J {@link Statement} as a Commons RDF {@link Quad}.
+	 * <p>
+	 * For the purpose of {@link BlankNode} equivalence, this 
+	 * method will use an internal salt UUID that is unique per instance of 
+	 * {@link RDF4JTermFactory}. 
+	 * <p>
+	 * <strong>NOTE:</strong> If combining RDF4J {@link Statement}s
+	 * multiple repositories or models, then their {@link BNode}s 
+	 * may have the same {@link BNode#getID()}, which with this method 
+	 * would become equivalent according to {@link BlankNode#equals(Object)} and
+	 * {@link BlankNode#uniqueReference()}, 
+	 * unless a separate {@link RDF4JTermFactory}
+	 * instance is used per RDF4J repository/model.  
+	 * 
+	 * @see #asQuad(Statement, UUID)
+	 * @param statement
+	 *            The statement to convert
+	 * @return A {@link RDF4JQuad} that is equivalent to the statement
+	 */
+	public RDF4JQuad asQuad(final Statement statement) {
+		return new QuadImpl(statement, salt);
```

Comment:
Yes, I think it would make more sense to hide the salt more and remove those static methods, perhaps have just a `RDF4JTermFactory(UUID)` constructor. Then the factory is not so locked into having to use UUID for salt.

_The reason to support a fixed `UUID` overall is if the user deliberately want them to be compatible - e.g. in a distributed architecture where two cloud instances are constructing named bnodes to be added to the same remote SAIL, or if you want reproducible output. (similar to fixed random seeds in simulations)._

I'll change it to just a constructor - removing the static methods will simplify the factory. 

In the [JenaRDFTermFactory](http://stain.github.io/incubator-commonsrdf/integration/org/apache/commons/rdf/jena/JenaRDFTermFactory.html) there's a different approach which I'm not sure what your views are on - the static methods there allow any `RDFTermFactory` to be passed in (it would use the `createBlankNode(String)` directly)